src/source/src/parser/pb_parser.rs

```diff
@@ -0,0 +1,305 @@
+            })
+        }
+        DataType::List {
+            datatype: item_type,
+        } => {
+            let value_list = value.and_then(|v| match v {
+                Value::Seq(l) => Some(l),
+                Value::Option(Some(boxed_l)) => match *boxed_l {
+                    Value::Seq(l) => Some(l),
+                    _ => None,
+                },
+                _ => None,
+            });
+            let item_schema = dtype_to_source_column_desc(item_type);
+            value_list.map(|values| {
+                let values = values
+                    .into_iter()
+                    .map(|value| from_protobuf_value(&item_schema, Some(value)))
+                    .collect::<Vec<Option<ScalarImpl>>>();
+                ScalarImpl::List(ListValue::new(values))
+            })
+        }
+        _ => unimplemented!(),
+    }
+}
+
+/// Maps protobuf type to RW type.
+fn protobuf_type_mapping(field_descriptor: &FieldDescriptor) -> Result<DataType> {
+    let field_type = field_descriptor.kind();
+    let mut t = match field_type {
+        Kind::Bool => DataType::Boolean,
+        Kind::Double => DataType::Float64,
+        Kind::Float => DataType::Float32,
+        Kind::Int32 | Kind::Sfixed32 | Kind::Fixed32 => DataType::Int32,
+        Kind::Int64 | Kind::Sfixed64 | Kind::Fixed64 => DataType::Int64,
+        Kind::String => DataType::Varchar,
+        Kind::Message(m) => {
+            let fields = m
+                .fields()
+                .map(|f| protobuf_type_mapping(&f))
+                .collect::<Result<Vec<_>>>()?;
+            let field_names = m.fields().map(|f| f.name().to_string()).collect::<Vec<_>>();
+            DataType::new_struct(fields, field_names)
+        }
+        // TODO(tabVersion): lack enum support
+        actual_type => {
+            return Err(NotImplemented(
+                format!("unsupported field type: {:?}", actual_type),
+                None.into(),
+            )
+            .into());
+        }
+    };
+    if field_descriptor.cardinality() == Cardinality::Repeated {
+        t = DataType::List {
+            datatype: Box::new(t),
+        }
+    }
+    Ok(t)
+}
+
+
+
+impl SourceParser for ProtobufParser {
+    fn parse(&self, payload: &[u8], writer: crate::SourceStreamChunkRowWriter<'_>) -> Result<WriteGuard> {
+        let message = DynamicMessage::decode(self.message_descriptor.clone(), payload).map_err(|e| ProtocolError(format!(
+            "parse message failed: {}", e
+        )))?;
+
+        writer.insert(|column_desc| {
+            let dyn_message = message.clone();
+            let value = dyn_message.get_field_by_name(column_desc.name.as_str());
+        })
+    }
+}
+
+mod test {
+    use super::*;
+    
+    #[test]
+    fn test_new_proto_parser() -> Result<()> {
+        let parser = ProtobufParser::new(
+            "file:///Users/tabversion/Desktop/risingwave/src/source/src/test_data/SCHEMA",
```

Comment:
Won't work on CI and other env?

Reply:
It's not ready yet. Just post to discuss some impl details